src/jit/lclvars.cpp

```diff
@@ -2159,17 +2159,24 @@ void               Compiler::lvaDecRefCnts(GenTreePtr tree)
 
     if ((tree->gtOper == GT_CALL) && (tree->gtFlags & GTF_CALL_UNMANAGED))
     {
-        /* Get the special variable descriptor */
+        if (opts.eeFlags & CORJIT_FLG_PINVOKE_USE_HELPERS)
+        {
+            noway_assert(info.compLvFrameListRoot == BAD_VAR_NUM);
+        }
+        else
+        {
+            /* Get the special variable descriptor */
 
-        lclNum = info.compLvFrameListRoot;
-            
-        noway_assert(lclNum <= lvaCount);
-        varDsc = lvaTable + lclNum;
-            
-        /* Decrement the reference counts twice */
+            lclNum = info.compLvFrameListRoot;
 
-        varDsc->decRefCnts(compCurBB->getBBWeight(this), this);  
-        varDsc->decRefCnts(compCurBB->getBBWeight(this), this);
+            noway_assert(lclNum <= lvaCount);
```

Comment:
This assert only executes once for each Unmanaged Calls, which is not frequently.